### PR TITLE
enhancement/series-area-3d

### PR DESCRIPTION
**Pull request by Izothep**
Added support for area series in 3D charts.

### CODE REVIEW

File: ts/Series/Area3DSeries.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        for (var i = 0; i < series.points.length; i++) {
+            bottomPoints.push({
+                x: series.rawPointsX[i],
+                y: options.stacking ? series.points[i].yBottom : translatedThreshold,
+                z: series.zPadding
+            } as any);
+        }
+    }
+
+    if (series.chart.is3d() && series.chart.options && series.chart.options.chart) {
```
Comment (pawelfus) [inline ts/Series/Area3DSeries.ts:74]: You checked `is3d()` already above

**Review by pawelfus — CHANGES_REQUESTED** [review]
Apart from that one minor thing, amazing work!

File: ts/Series/Area3DSeries.ts (hunk 1 above)
Comment (Izothep) [reply ts/Series/Area3DSeries.ts]: I have removed this check. Thanks!

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks, it looks great!

It seems the Z index is not 100% though: https://jsfiddle.net/highcharts/dk486t5m/

In addition I think we need some more visual samples both for user education and for regression testing:
* Multiple series non-stacked
* Multiple series with stacking
* Perhaps a combination of stacking and not

**Comment by Izothep** [comment]
All corrected, thanks @TorsteinHonsi! I will add the demo to the related docs article in a separate PR.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

There is still a Z index issue with the markers: https://jsfiddle.net/highcharts/dkqe7m5y/. Note how _Series 2_'s markers are hidden behind _Series 1_'s fill area.

> I will add the demo to the related docs article in a separate PR.

Can we have the docs and the nice demo in the same PR please? So that we don't lose track of this new feature.

**Comment by Izothep** [comment]
Thanks for the review. `zIndex` problem is solved in the newest version of the code. I have added a demo and prepared docs explanation (I am not sure if they are written correctly though).

**Review by TorsteinHonsi — APPROVED** [review]
Perfect, thanks!

**Pull request merged**